src/qibolab/instruments/zhinst.py

```diff
@@ -107,201 +115,210 @@ def select_pulse(pulse, pulse_type):
-                    start=0,
-                    duration=sequence.duration + sequence.start,
-                    amplitude=1,
-                    shape="Rectangular",
-                    channel=qubit.flux.name,
-                    qubit=qubit.name,
-                )
-                self.signal = f"flux{qubit.name}"
-            if isinstance(qubit, Coupler):
-                pulse = CouplerFluxPulse(
-                    start=0,
-                    duration=sequence.duration + sequence.start,
-                    amplitude=1,
-                    shape="Rectangular",
-                    channel=qubit.flux.name,
-                    qubit=qubit.name,
-                )
-                self.signal = f"couplerflux{qubit.name}"
+    def sweeps_for_channel(self, ch: str) -> list[tuple[Parameter, lo.SweepParameter]]:
+        return [item[1:] for item in self._channel_sweeps if item[0] == ch]
 
-            self.pulse = pulse
+    def sweeps_for_sweeper(self, sweeper: Sweeper) -> list[lo.SweepParameter]:
+        return [item[1] for item in self._parallel_sweeps if item[0] == sweeper]
 
-            self.zhpulse = lo.pulse_library.const(
-                uid=(f"{pulse.type.name.lower()}_{pulse.qubit}_"),
-                length=round(pulse.duration * NANO_TO_SECONDS, 9),
-                amplitude=pulse.amplitude,
-            )
+    def channel_sweeps_for_sweeper(
+        self, sweeper: Sweeper
+    ) -> list[tuple[str, Parameter, lo.SweepParameter]]:
+        return [
+            item
+            for item in self._channel_sweeps
+            if item[2] in self.sweeps_for_sweeper(sweeper)
+        ]
 
-        elif sweeper.parameter is Parameter.start:
-            if pulse:
-                self.pulse = pulse
-                self.signal = f"flux{qubit}"
+    def channels_with_sweeps(self) -> set[str]:
+        return {ch for ch, _, _ in self._channel_sweeps}
 
-                self.zhpulse = ZhPulse(pulse).zhpulse
 
-        # Need something better to store multiple sweeps on the same pulse
-        self.zhsweeper = self.select_sweeper(sweeper)
+@dataclass
+class SubSequence:
+    """A subsequence is a slice (in time) of a sequence that contains at most
+    one measurement per qubit.
+
+    When the driver is asked to execute a sequence, it will first split
+    it into sub-sequences. This is needed so that we can create a
+    separate laboneq section for each measurement (multiple measurements
+    per section are not allowed). When splitting a sequence, it is
+    assumed that 1. a measurement operation can be parallel (in time) to
+    another measurement operation (i.e. measuring multiple qubits
+    simultaneously), but other channels (e.g. drive) do not contain any
+    pulses parallel to measurements, 2. ith measurement on some channel
+    is in the same subsequence as the ith measurement (if any) on
+    another measurement channel, 3. all measurements in one subsequence
+    happen at the same time.
```

Comment:
I didn't introduce these assumptions with this PR. They were already there in various places in the code. I have just collected them in on place and documented.